### PR TITLE
Fix sedit command, mismatch of stall name for reviews

### LINKED ISSUE #199: Fix sedit command, mismatch of stall name for reviews
When the sedit command to change the name of a stall is run, the stall name for the reviews under the edited stalls are not changed.

**Before the sedit command is run:**
image

**After the sedit command is run:**
image

### PR DESCRIPTION
**Pull request by MinHeinA**
Resolve #199 

When the sedit command changes the stall name, the stall names in the reviews associated with the stalls are not modified in both the review list in model and stall. 

The sedit now updates the stall name of the associated reviews in model and stall.

Updating associated reviews in model:
image

Updating associated reviews in stall:
image

### CODE REVIEW

File: src/main/java/foodwhere/logic/commands/SEditCommand.java [hunk 1]
```diff
@@ -87,7 +103,25 @@ private static Stall createEditedStall(Stall stallToEdit, EditStallDescriptor ed
         Address updatedAddress = editStallDescriptor.getAddress().orElse(stallToEdit.getAddress());
         Set<Tag> updatedTags = editStallDescriptor.getTags().orElse(stallToEdit.getTags());
 
-        return new Stall(updatedName, updatedAddress, updatedTags);
+        Set<Review> reviewsToUpdate = stallToEdit.getReviews();
```
Comment (clarence-chew) [inline src/main/java/foodwhere/logic/commands/SEditCommand.java:106]: I think it seems a bit strange to do reviews editing outside the EditStallDescriptor object, why not edit the set of reviews inside EditStallDescriptor as well?

File: src/main/java/foodwhere/model/stall/Stall.java [hunk 2]
```diff
@@ -72,6 +72,14 @@ public Set<Review> getReviews() {
         return Collections.unmodifiableSet(reviews);
     }
 
+    /**
```
Comment (clarence-chew) [inline src/main/java/foodwhere/model/stall/Stall.java:75]: This makes the stall mutable, which makes it even more costly to implement `undo`. Immutable stalls are also better for defensive programming.

**Review by clarence-chew — COMMENTED** [review]
Here's some comments on the update.

**Comment by MinHeinA** [comment]
Alternate and better fix by @clarence-chew in PRs #204 , #206 , #207 and #211

**Pull request closed without merging**